Add Light Hotspot and General Hotspot Related Content Reorg (#472)
helium/docs: docs/mine-hnt/data-only-hotspots.mdx

---
id: data-only-hotspots
title: Data Only Hotspots
sidebar_label: Data Only Hotspots
slug: /mine-hnt/data-only-hotspots
---

# Data Only Hotspots

import useBaseUrl from "@docusaurus/useBaseUrl";

:::warning
Adding Data Only Hotspots is not yet enabled on chain, but is expected to be

> released in early July. This guide is released for preparation only.

Reviewer: ```suggestion
released in early July. Earning $HNT for data transfer is expected to be enabled at a later time as part of Milestone 2. Please look for updates in #gateway-development channel. This guide is released for preparation only.
```

:::

Data Only Hotspots are eligible for **Network Data Transfer Rewards** but not
**Proof of Coverage Rewards**.

> Since Data Only Hotspots do not earn Proof of Coverage rewards, they may be

Reviewer: ```suggestion
Since Data Only Hotspots do not earn Proof-of-Coverage rewards, they may be
```

permissionlessly added to the blockchain. Once added to the blockchain and
with location asserted, they will earn $HNT when forwarding network data.

> # Setup a data only hotspot

Reviewer: ```suggestion
# Setup a Data Only Hotspot
```

## Transactions & Cost

> Every hotspot on the blockchain goes through two transactions:

Reviewer: ```suggestion
Every Hotspot on the blockchain goes through two transactions:
```

> * **add hotspot**: links the hotspot's key (`swarm_key`) of a hotspot to a

Reviewer: ```suggestion
* **add hotspot**: links the Hotspot's key (`swarm_key`) of a Hotspot to a
```

user's wallet

> * **assert location**: provides GPS location, altitude, and dBi antenna

Reviewer: ```suggestion
* **assert location**: provides GPS location, elevation (in meters), and dBi antenna
```

> output for the hotspot

Reviewer: ```suggestion
output for the Hotspot
```

The fee schedule is as follows:
* **add data only hotspot**: 500000 DC ($5)
* **assert location**: 1000000 DC ($10)

## Pre-requisites

You should [setup a packet forwarder](/use-the-network/setup-a-packet-forwarder)
with the [light hotspot client](/use-the-network/light-hotspots).

In addition, you should setup a [CLI wallet](/wallets/cli-wallet) as this

> is currently the only wallet that supports data only hotspots.

Reviewer: ```suggestion
is currently the only wallet that supports Data Only Hotspots.
```

**Note**: you can import a mobile wallet to the CLI.
**Another note**: you should __not__ install the CLI on your gateway

## Add Hotspot

> From a terminal on the device where the light hotspot client is installed,

Reviewer: ```suggestion
From a terminal on the device where the Light Hotspot client is installed,
```

  "payer": "14GWyFj9FjLHzoN3aX7Tq7PL6fEg4dfWPY8CrK8b9S5ZrcKDz6S",
  "staking fee": 500000,
  "txn": "CrkBCiEBrlImpYLbJ0z0hw5b4g9isRyPrgbXs9X+RrJ4pJJc9MkSIQA7yIy7F+9oPYCTmDz+v782GMJ4AC+jM+VfjvUgAHflWSJGMEQCIGfugfLkXv23vJcfwPYjLlMyzYhKp+Rg8B2YKwnsDHaUAiASkdxUO4fdS33D7vyid8Tulizo9SLEL1lduyvda9YVRCohAa5SJqWC2ydM9IcOW+IPYrEcj64G17PV/kayeKSSXPTJOMCEPUDo+wM="
}
```

The `txn` field needs to be used as the input to the wallet command
`helium-wallet hotspot add`. For example:

```
helium-wallet hotspots add CrkBCiEBrlImpYLbJ0z0hw5b4g9isRyPrgbXs9X+RrJ4pJJc9MkSIQA7yIy7F+9oPYCTmDz+v782GMJ4AC+jM+VfjvUgAHflWSJGMEQCIGfugfLkXv23vJcfwPYjLlMyzYhKp+Rg8B2YKwnsDHaUAiASkdxUO4fdS33D7vyid8Tulizo9SLEL1lduyvda9YVRCohAa5SJqWC2ydM9IcOW+IPYrEcj64G17PV/kayeKSSXPTJOMCEPUDo+wM=
```

After you verify the transaction, use append the command with `--commit`
to submit it to the API. Note that you need sufficient funds to cover
both the transaction cost (65000 DC) and the `staking_fee` (500000 DC).
Thanks to the ["implicit burn" mechanism](/blockchain/transaction-fees/#transaction-fees-and-implicit-burn),
available HNT will be used to cover the transaction if you have
insufficient DCs.

> You can track the transactions progress in being processed by using the

Reviewer: ```suggestion
You can track the transaction progress by using the
```

transaction hash at the following endpoint: `https://api.helium.io/v1/pending_transactions/TXN_HASH`

> Once the transaction clears, you're done linking the hotspot with your

Reviewer: ```suggestion
Once the transaction clears, you're done linking the Hotspot with your
```

wallet.

:::warning
At this point, you will want to be certain that you've backed up the

> light hotspot's keypair. It's location can be customized in the client's

Reviewer: ```suggestion
light hotspot's keypair. Its location can be customized in the client's
```

`settings.toml`, but by default, it's located at
`/etc/helium_gateway/key.bin`
:::

## Assert Hotspot

> To assert the gateway using the `helium-wallet hotspots assert`

Reviewer: ```suggestion
To assert the Hotspot using the `helium-wallet hotspots assert`
```

```
helium-wallet hotspots assert --gateway LIGHT_GATEWAY_PUBKEY --lat=37.7806295 --lon=-122.4008481
```

The `LIGHT_GATEWAY_PUBKEY` in the example from the previous section
would be `11TL62V8NYvSTXmV5CZCjaucskvNR1Fdar1Pg4Hzmzk5tk2JBac`.

> After you verify the transaction, use append the command with `--commit`

Reviewer: ```suggestion
After you verify the transaction, append the command with `--commit`
```

to submit it to the API. Note that you need sufficient funds to cover
both the transaction cost (65000 DC) and the `staking_fee` (1000000 DC).
Thanks to the ["implicit burn" mechanism](/blockchain/transaction-fees/#transaction-fees-and-implicit-burn),
available HNT will be used to cover the transaction if you have
insufficient DCs.

> You can track the transactions progress in being processed by using the

Reviewer: ```suggestion
You can track the transaction progress by using the
```

transaction hash at the following endpoint: `https://api.helium.io/v1/pending_transactions/TXN_HASH`

> Once the transaction clears, you're data only hotspot is ready to start

Reviewer: ```suggestion
Once the transaction clears, your Data Only Hotspot is ready to start
```
